src/charm.py

```diff
@@ -82,78 +86,108 @@ def __init__(self, *args):
-        self.framework.observe(self._database.on.database_created, self._configure_db)
-        self.framework.observe(self._database.on.endpoints_changed, self._configure_db)
+        self.framework.observe(self._database.on.database_created, self._configure_webui)
+        self.framework.observe(self._auth_database.on.database_created, self._configure_webui)
+        self.framework.observe(self._database.on.endpoints_changed, self._configure_webui)
+        self.framework.observe(self._auth_database.on.endpoints_changed, self._configure_webui)
         self.framework.observe(
             self.on.sdcore_management_relation_joined, self._publish_sdcore_management_url
         )
         # Handling config changed event to publish the new url if the unit reboots and gets new IP
         self.framework.observe(self.on.config_changed, self._publish_sdcore_management_url)
 
-    def _configure(self, event: EventBase) -> None:
-        """Juju event handler.
+    def _configure_webui(self, event: EventBase) -> None:
+        """Main callback function of the Webui operator.
 
-        Sets unit status, writes configuration file adds pebble layer.
+        Handles config changes.
+        Manages pebble layer and Juju unit status.
 
         Args:
-            event (EventBase): Juju event.
+            event: Juju event
         """
-        if not self._database_relation_is_created():
-            self.unit.status = BlockedStatus("Waiting for database relation to be created")
+        for relation in ["database", "auth_database"]:
+            if not self._relation_created(relation):
+                self.unit.status = BlockedStatus(f"Waiting for {relation} relation to be created")
+                return
+        if not self._database_is_available():
+            self.unit.status = WaitingStatus("Waiting for the database to be available")
+            return
+        if not self._auth_database_is_available():
+            self.unit.status = WaitingStatus("Waiting for the database to be available")
             return
         if not self._container.can_connect():
             self.unit.status = WaitingStatus("Waiting for container to be ready")
+            event.defer()
             return
         if not self._container.exists(path=BASE_CONFIG_PATH):
             self.unit.status = WaitingStatus("Waiting for storage to be attached")
+            event.defer()
             return
+
+        config_file_content = render_config_file(
+            database_name=DEFAULT_DATABASE_NAME,
+            database_url=self._get_database_data()["uris"].split(",")[0],
+            auth_database_name=AUTH_DATABASE_NAME,
+            auth_database_url=self._get_auth_database_data()["uris"].split(",")[0],
+        )
+        self._write_config_file(content=config_file_content)
         if not self._config_file_exists():
             self.unit.status = WaitingStatus("Waiting for config file to be written")
             return
```

Comment:
```suggestion
```
since the config file is always written just above

Reply:
Removed.